### PR TITLE
MM-50122 : Remove and test refetching of channels on browser focus change

### PR DESCRIPTION
**Pull request by M-ZubairAhmed**
#### Ticket Link
https://mattermost.atlassian.net/browse/MM-50122

#### Screenshots
n/a

#### Release Note

```release-note
Added experimental feature to disable re-fetching of channel and channel members on browser focus.
```

### CODE REVIEW

**Comment by M-ZubairAhmed** [comment]
Dev review was completed in older PR, so only adding QA after a quick review from @hmhealey

**Review by yasserfaraazkhan — APPROVED** [review]

**Comment by streamer45** [comment]
Very interesting. We were discussing this same logic very recently with @agarciamontoro and @agnivade. I think we should update the load-test client to follow this flag and see the difference it makes.

**Comment by agnivade** [comment]
This looks interesting. But I am not a huge fan of introducing more flags and knobs for the user to control. Why not make this the default behavior? And if we want to gradually deploy this, a feature flag seems ideal.

**Review by isacikgoz — APPROVED** [review]

**Comment by isacikgoz** [comment]
I'm fine with both approaches (feature flag vs experimental config), but a load-test results would be good.

**Comment by agarciamontoro** [comment]
Oh, this is interesting! We were indeed discussing about removing the simulation of this action altogether from the load-test. Whatever we end up doing, feature flag or config setting, we should port the change to the load-test code, yes. @M-ZubairAhmed, would you be open to do that?

**Comment by M-ZubairAhmed** [comment]
> Oh, this is interesting! We were indeed discussing about removing the simulation of this action altogether from the load-test. Whatever we end up doing, feature flag or config setting, we should port the change to the load-test code, yes. @M-ZubairAhmed, would you be open to doing that?

I would be open to that; let's chat on Community about how I can do that. Do we have any reference docs I can read up before?

**Comment by M-ZubairAhmed** [comment]
@yasserfaraazkhan any failures in E2E tests i should be worried about?

**Comment by yasserfaraazkhan** [comment]
> @yasserfaraazkhan any failures in E2E tests i should be worried about?

@M-ZubairAhmed , lot of those are known ones. nothing caused by this PR

File: webapp/channels/src/components/team_controller/team_controller.tsx [hunk 1]
```diff
@@ -70,7 +70,7 @@ function TeamController(props: Props) {
         const wakeUpIntervalId = setInterval(() => {
             const currentTime = Date.now();
             if ((currentTime - lastTime.current) > WAKEUP_THRESHOLD) {
-                console.log('computer woke up - fetching latest'); //eslint-disable-line no-console
+                // If the user has been inactive for more than WAKEUP_THRESHOLD, try to reconnect the websocket
```
[the remark this answers is not in the document]
Comment (M-ZubairAhmed) [reply webapp/channels/src/components/team_controller/team_controller.tsx:73]: reverted

File: webapp/channels/src/components/team_controller/team_controller.tsx [hunk 2]
```diff
@@ -92,12 +92,15 @@ function TeamController(props: Props) {
...
             }
 
-            const currentTime = Date.now();
-            if ((currentTime - blurTime.current) > UNREAD_CHECK_TIME_MILLISECONDS && props.currentTeamId) {
-                if (props.graphQLEnabled) {
-                    props.fetchChannelsAndMembers(props.currentTeamId);
-                } else {
-                    props.fetchMyChannelsAndMembersREST(props.currentTeamId);
+            // Temporary flag to disable refetching of channel members on browser focus
+            if (props.disableRefetchingOnBrowserFocus === false) {
```
[the remark this answers is not in the document]
Comment (M-ZubairAhmed) [reply webapp/channels/src/components/team_controller/team_controller.tsx:96]: done

**Pull request merged**